server/player/player.go

```diff
@@ -1911,8 +1911,34 @@ func (p *Player) EyeHeight() float64 {
 
 // PlaySound plays a world.Sound that only this Player can hear. Unlike World.PlaySound, it is not broadcast
 // to players around it.
-func (p *Player) PlaySound(sound world.Sound) {
-	p.session().ViewSound(entity.EyePosition(p), sound)
+// if passing world.Sound argument it should be use LevelSoundEvent, if passing string argument it should be use PlaySound
+func (p *Player) PlaySound(sound ...interface{}) {
+	switch so := sound[0].(type) {
+	case world.Sound:
+		p.session().ViewSound(entity.EyePosition(p), so)
+	case string:
```

Comment:
Just implement world.Sound with a new type. I wouldn't add a special case for this.

Reply:
Like world.PlaySound ?

Reply:
Just implement it like any other sound is currently implemented. Make it a struct in the world/sound package and have fields in it like the name of the particle, pitch etc. Then make session send it appropriately in the `ViewSound` method.

Reply:
OK